rename.g

```diff
@@ -0,0 +1,324 @@
+## 
+## First add and commit the renamings_to_execute.csv file. -- Commit 1
+## Then discuss on github if the suggested names are suitable.
+## After approval run rename.g from within the package directory.
+## Run tst/testquick.g or tst/testall.g and ensure everything is still working.
+## On failure you can always roll back to Commit 1, i.e. the version before 
+## applying the renamings. 
+## On success add and commit the changed files. -- Commit 2
+##
+#############################################################################
+
+DeclareSynonym("Yes", true);
+DeclareSynonym("No", false);
+
+GetInputFromUser := function(message)
+local input;
+
+    input := InputFromUser(message);
+    if input = No then
+        return false;
+    elif input = Yes then
+        return true;
+    else
+        Print("Input is not valid.\n");
+        return GetInputFromUser(message);
+    fi;
+end;
+
+# Returns true on failed check. Helper method of CheckRenamingTable.
+# Check fails if there exist duplicate entries in colA and colB.
+# If isFirstInHistory is true, then colA is from the renamings_history.csv file,
+# otherwise from the renamings_to_execute.csv file.
+# colB is always from the renamins_to_execute.csv file.
+CheckRenamingTableForDuplicates := function(colA, colB, colNameA, colNameB, isFirstInHistory, message)
+local failedCheck, toTest, word, posA, posB, fileA, fileB;
+    
+    if colNameA = colNameB and isFirstInHistory = false then
+        toTest := Filtered(DuplicateFreeList(colA), x -> Number(colA, y -> y = x) > 1);
+    else
+        toTest := Intersection(colA, colB);
+    fi;
+    failedCheck := not IsEmpty(toTest);
+    while not IsEmpty(toTest) do
+        word := Remove(toTest);
+        if isFirstInHistory then
+            posA := First([1..Length(colA)], i -> colA[i] = word);
+            fileA := "renamings_history.csv";
+            posB := First([1..Length(colB)], i -> colB[i] = word);
+            fileB := "renamings_to_execute.csv";
+        else
+            posA := First([1..Length(colA)], i -> colA[i] = word);
+            fileA := "renamings_to_execute.csv";
+            if colNameA = colNameB then
+                posB := First([1..Length(colB)] + posA, i -> colB[i] = word);
+                fileB := "renamings_to_execute.csv";
+            else
+                posB := First([1..Length(colB)], i -> colB[i] = word);
+                fileB := "renamings_to_execute.csv";
+            fi;
+        fi;
+
+        Print(message,"\n\tThe word \'",word,"\' is located \n\tin line ",posA+1," in column \'",colNameA,"\' in file \'",fileA,"\' and\n\tin line ",posB+1," in column \'",colNameB,"\' in file \'",fileB,"\'\n");
+    od;
+    return failedCheck;
+end;
+
+# Returns true on failed check. Helper method of CheckRenamingTable.
+# Check fails if grep returns failureInt on search of a word in col.
+# failureInt is either 0 or 1.
+CheckExistenceOfWord := function(col, colName, failureInt, message)
```

Comment:
You can let the function return whether  a `word` was found and remove the `failureInt` variable. That makes the function easier to understand.

Reply:
Yes, but then I would need to store somewhere the words which are (found/not found) for the error message, which is printed at the moment from within the function depending on the `failureInt`. I think this would be even less readable.

Reply:
Ah, true. You could in principle return the list of found words and let the rest be handled by the calling function. But let's not get too concerned with details. Leave the function like it is if you want to.